src/Bicep.Local.Deploy.IntegrationTests/ProviderExtensionTests.cs

```diff
@@ -54,11 +54,9 @@ await Task.WhenAll(
 
                 builder.AddBicepExtensionHost(processArgs);
                 registerHandlers(builder.Services
-                    .AddBicepExtension(
-                        name: "MockExtension",
-                        version: "0.0.1",
-                        isSingleton: true,
-                        typeAssembly: typeof(ProviderExtensionTests).Assembly));
+                    .AddBicepExtension()
```

Comment:
The changes look great to me - only things I would want to call out:
* Looks like you are hitting some test failures
* Can we retain the old code path with an `[Obsolete]` attribute, to make the transition for existing users simpler?